src/statistic/update.py

```diff
@@ -273,53 +269,54 @@ def get_architecture_stats(self):
             pipeline_match={'processed_analysis.cpu_architecture.summary': {'$exists': True, '$not': {'$size': 0}}},
             additional_projection={'processed_analysis.cpu_architecture.summary': 1}
         )
-        result = [self._shorten_architecture_string(self._find_most_frequent_architecture(list(itertools.chain.from_iterable(item['architecture']))))
-                  for item in self.db.file_objects.aggregate(aggregation_pipeline)]
+        query_result = self.db.file_objects.aggregate(aggregation_pipeline, allowDiskUse=True)
+        result = [
+            self._shorten_architecture_string(self._find_most_frequent_architecture(list(itertools.chain.from_iterable(item['architecture']))))
+            for item in query_result
+        ]
         return {'cpu_architecture': self._count_occurrences(result)}
 
     def _find_most_frequent_architecture(self, arch_list):
         try:
             arch_frequency = sorted(self._count_occurrences(arch_list), key=lambda x: x[1], reverse=True)
             return arch_frequency[0][0]
-        except Exception as e:
-            logging.error('Could not get arch frequency: {} {}'.format(sys.exc_info()[0].__name__, e))
+        except (AttributeError, KeyError, TypeError) as exception:
+            logging.error('Could not get arch frequency: {} {}'.format(sys.exc_info()[0].__name__, exception))
             return None
 
     @staticmethod
-    def _count_occurrences(l):
-        return [(item, l.count(item)) for item in set(l)]
+    def _count_occurrences(result_list):
+        return [(item, result_list.count(item)) for item in set(result_list)]
 
     @staticmethod
-    def _shorten_architecture_string(s):
-        if s is None:
+    def _shorten_architecture_string(string):
```

Comment:
I generally try to omit `string` as identifier since there is the built-in library is called `string`. But I guess it's fair game as long as you don't need that..